lld/MachO/Driver.cpp

```diff
@@ -1295,6 +1394,27 @@ static void createFiles(const InputArgList &args) {
       break;
     }
   }
+
+  if (config->readThreads) {
+    multiThreadedPageIn(deferredFiles);
+
+    DeferredFiles archiveContents;
+    std::vector<ArchiveFile *> archives;
+    for (auto &file : deferredFiles)
+      if (ArchiveFile *archive = dyn_cast<ArchiveFile>(
+              processFile(file.buffer, &archiveContents, file.path,
+                          LoadType::CommandLine, file.isLazy)))
+        archives.push_back(archive);
+
+    if (!archiveContents.empty()) {
+      multiThreadedPageIn(archiveContents);
+      for (auto *archive : archives)
+        archive->addLazySymbols();
+    }
```

Comment:
A couple of things about this:
I think `addLazySymbols()` will happen at the same time that `multiThreadedPageIn` is loading the contents from the archives. I don't know if that is intentional or if it is fine to ignore.

Not sure if one has to do `multiThreadedPageIn` for the `archiveContents`. I might be understanding this incorrectly, but if `ArchiveFile` is a `.a`, wouldn't the first `multiThreadedPageIn(deferredFiles)` already had loaded them?

The second thing is that the handling of the archive contents makes everything very convoluted. Using the presence of the argument `archiveContents` as a flag to add the contents to it, AND to perform or not the `addLazySymbols` is complicated to understand (and I think it is easier when I am looking only at this PR).

Reply:
Forgot about thin archives. If they share their representation with "thick" archives, unless more complexity is added, it will double page in those "thick" archives, but that should be fine. Besides the threading complexity, the loop for paging in should be quick.

I wonder how things work with `--start-lib`, `--end-lib` in the command line. Not sure how those end up being represented.